Ical.Net/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -483,7 +484,7 @@ private List<DateTime> GetYearDayVariants(List<DateTime> dates, RecurrencePatter
                         ? date.AddDays(-date.DayOfYear + yearDay)
                         : date.AddDays(-date.DayOfYear + 1).AddYears(1).AddDays(yearDay);
 
-                    if (newDate.DayOfYear == date.DayOfYear)
+                    if (newDate.Date == date.Date)
```

Comment:
The outer loop has no code coverage right now, assume this will follow after merging #617 get merged

Reply:
Yes, this was the idea :-)